### ISSUE TITLE
Parse error with tilde character

### ISSUE THREAD

**Issue by athom370-ford** [JohnnyMorganz/StyLua#71]
### TL;DR
`~` breaks the StyLua VS Code extension.

### Background
[Lua 5.3](https://www.lua.org/manual/5.3/manual.html#3.4.2) supports the unary bitwise NOT operator (`~`). I installed the StyLua VS Code extension and followed [these instructions](https://github.com/JohnnyMorganz/StyLua#using-the-vscode-extension) to set it up.

### Reproduction Steps
1. In VS Code, save a `.lua` file containing a `~` in an expression.
2. See a pop-up error in the bottom right corner of VS Code:  
`Could not format file: error: could not format from stdin: error parsing: error occurred while tokenizing: unexpected character * at line X, column Y`
X and Y correspond to the exact location of the `~`. VS Code lists the source of the error as `StyLua (Extension)`. 

#### Is this a valid issue? If so, can it be fixed?

**Comment by JohnnyMorganz** [comment]
Hi @athom370-ford 
StyLua uses [https://github.com/Kampfkarren/full-moon](full-moon) for parsing, and currently it only supports Lua 5.1 (and Luau).
You will need to open up an issue in the full-moon repository for Lua 5.3 syntax support - if its completed, I'm happy to support it in StyLua.
